tools/c7n_azure/c7n_azure/policy.py

```diff
@@ -50,50 +59,50 @@ class AzureFunctionMode(ServerlessExecutionMode):
     def __init__(self, policy):
         self.policy = policy
         self.log = logging.getLogger('custodian.azure.AzureFunctionMode')
-        s = local_session(self.policy.session_factory)
-        self.client = s.client('azure.mgmt.web.WebSiteManagementClient')
+        self.session = local_session(self.policy.session_factory)
+        self.client = self.session.client('azure.mgmt.web.WebSiteManagementClient')
+
+        self.template_util = TemplateUtilities()
+        self.parameters = self._get_parameters(self.template_util)
+        self.group_name = self.parameters['servicePlanName']['value']
+        self.webapp_name = self.parameters['name']['value']
+        self.policy_name = self.policy.data['name'].replace(' ', '-').lower()
 
     def run(self, event=None, lambda_context=None):
         """Run the actual policy."""
         raise NotImplementedError("subclass responsibility")
 
     def provision(self):
         """Provision any resources needed for the policy."""
-        template_util = TemplateUtilities()
-
-        parameters = self._get_parameters(template_util)
-        group_name = parameters['servicePlanName']['value']
-        webapp_name = parameters['name']['value']
-        policy_name = self.policy.data['name'].replace(' ', '-').lower()
-
         existing_service_plan = self.client.app_service_plans.get(
-            group_name, parameters['servicePlanName']['value'])
+            self.group_name, self.parameters['servicePlanName']['value'])
```

Comment:
`self.parameters['servicePlanName']['value'] == self.group_name` right?

Reply:
Yeah but I don't think it's very clear if we just send in group_name, group_name....

Reply:
Also if we ever change the code for how we set group name then this could break.